src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/RoleControllerIntTest.kt

```diff
@@ -3,14 +3,296 @@ package uk.gov.justice.digital.hmpps.externalusersapi.resource
+          assertThat(it["userMessage"] as String).contains("default message [roleName],128,4]")
+        }
+    }
+
+    @Test
+    fun `Create role length too long`() {
+      webTestClient
+        .post().uri("/api/roles")
+        .headers(setAuthorisation("AUTH_ADM", listOf("ROLE_ROLES_ADMIN")))
+        .body(
+          BodyInserters.fromValue(
+            mapOf(
+              "roleCode" to "x".repeat(30) + "x",
+              "roleName" to "x".repeat(128) + "y",
+              "roleDescription" to "x".repeat(1024) + "y",
+              "adminType" to listOf("EXT_ADM")
+            )
+          )
+        )
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String).startsWith("Validation failure:")
+          assertThat(it["userMessage"] as String).contains("default message [roleCode],30,2]")
+          assertThat(it["userMessage"] as String).contains("default message [roleName],128,4]")
+          assertThat(it["userMessage"] as String).contains("default message [roleDescription],1024,0]")
+        }
+    }
+
+    @Test
+    fun `Create role admin type empty`() {
+      webTestClient
+        .post().uri("/api/roles")
+        .headers(setAuthorisation("AUTH_ADM", listOf("ROLE_ROLE_ADMIN")))
+        .body(
+          BodyInserters.fromValue(
+            mapOf(
+              "roleCode" to "xxxx", "roleName" to "123456", "adminType" to listOf<String>()
+            )
+          )
+        )
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String)
+            .startsWith("Validation failure:")
+          assertThat(it["userMessage"] as String)
+            .contains("default message [Admin type cannot be empty]")
+        }
+    }
+
+    @Test
+    fun `Create role failed regex`() {
+      webTestClient
+        .post().uri("/api/roles")
+        .headers(setAuthorisation("AUTH_ADM", listOf("ROLE_ROLES_ADMIN")))
+        .body(
+          BodyInserters.fromValue(
+            mapOf(
+              "roleCode" to "a-b",
+              "roleName" to "a\$here",
+              "roleDescription" to "a\$description",
+              "adminType" to listOf("EXT_ADM")
+            )
+          )
+        )
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String)
+            .contains("default message [roleCode],[Ljavax.validation.constraints.Pattern")
+          assertThat(it["userMessage"] as String)
+            .contains("default message [roleName],[Ljavax.validation.constraints.Pattern")
+          assertThat(it["userMessage"] as String)
+            .contains("default message [roleDescription],[Ljavax.validation.constraints.Pattern")
+        }
```

Comment:
There are no tests for developerMessage - I'm assuming this is because it's virtually the same as usermessage?

Reply:
no it's fine